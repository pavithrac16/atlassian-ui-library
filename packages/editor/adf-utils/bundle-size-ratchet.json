[
  {
    "team": "editor",
    "package": "@atlaskit/adf-utils",
    "version": "6.1.0",
    "dependent": true,
    "id": "src.main",
    "name": "main",
    "stats": {
      "size": 31368,
      "gzipSize": 6696
    }
  },
  {
    "team": "editor",
    "package": "@atlaskit/adf-utils",
    "version": "6.1.0",
    "dependent": true,
    "id": "combined.main",
    "name": "main",
    "stats": {
      "size": 31368,
      "gzipSize": 6696
    }
  }
]