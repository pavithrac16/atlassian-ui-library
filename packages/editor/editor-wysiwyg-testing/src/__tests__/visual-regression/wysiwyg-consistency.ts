import {
  createDocumentADF,
  loadKitchenSinkWithAdf,
  snapshotAndCompare,
} from './_wysiwyg-utils';
import { waitForLoadedBackgroundImages } from '@atlaskit/visual-regression/helper';
import { traverse } from '@atlaskit/adf-utils/traverse';

type AsyncAwaitFunction = (page: any) => Promise<void>;
type FragmentNodeLookup = { node: string; waitFor?: AsyncAwaitFunction[] };

describe('WYSIWYG Snapshot Test: looks consistent in editor & renderer', () => {
  let page: any;

  /**
   * ADF Node Fragments for WYSIWYG consistency testing.
   *
   * Note: these node names should match the de-hyphenated name of the adf fragment json
   * files contained within `__fixtures__/adf-node-fragments`.
   *
   * We declare this explicit list instead of reading the directory's files via the file system,
   * as this gives us greater control to skip specific nodes, should the need ever arise.
   *
   * Because these tests need to be deterministic, if a node requires additional time to load
   * a resource you can add a `waitForSelector` value to defer the screenshot until it's available.
   */
  const contentNodes: FragmentNodeLookup[] = [
    { node: 'actions' },
    { node: 'blockquote' },
    { node: 'bullet list' },
    { node: 'codeblock' },
    { node: 'date' },
    { node: 'decisions' },
    { node: 'divider' },
    {
      node: 'emoji',
      waitFor: [
        async (page: any): Promise<void> => {
          const emojiSelector = '.emoji-common-emoji-sprite';
          await page.waitForSelector(emojiSelector);
          await waitForLoadedBackgroundImages(page, emojiSelector);
        },
      ],
    },
    { node: 'heading' },
    { node: 'mention' },
    { node: 'ordered list' },
    { node: 'paragraph' },
    { node: 'status' },
  ];

  const containerNodes: FragmentNodeLookup[] = [
    { node: 'table' },
    { node: 'columns' },
    { node: 'panel' },
  ];

  beforeAll(async () => {
    // @ts-ignore
    page = global.page;
    await page.setViewport({ width: 2000, height: 1000 });
  });

  describe('Standalone content', () => {
    // Here we filter out nodes that don't render anything in their initial state
    const standaloneNodes = contentNodes.concat(
      containerNodes.filter(
        (node: FragmentNodeLookup) => node.node !== 'columns',
      ),
    );

    it.each(standaloneNodes)('%p', async ({ node, waitFor }) => {
      const adf = createDocumentADF(node);
      await loadKitchenSinkWithAdf(page, adf);
      await snapshotAndCompare(page, node, waitFor);
    });
  });

  describe('Nested content', () => {
    describe.each(containerNodes)(
      '',
      ({ node: containerNode }: FragmentNodeLookup) => {
        const containerAdf = createDocumentADF(containerNode, true);
        // Nested nodes
        it.each(contentNodes)(
          `%p inside ${containerNode}`,
          async ({ node: contentNode, waitFor }: FragmentNodeLookup) => {
            const contentAdf = createDocumentADF(contentNode).content;
            const adf = traverse(containerAdf, {
              any: (node: any) => {
                if (node.content && node.content.length === 0) {
                  // Insert nested content into container
                  node.content = contentAdf;
                }
                return node;
              },
            });
            await loadKitchenSinkWithAdf(page, adf);
            await snapshotAndCompare(
              page,
              `${contentNode} inside ${containerNode}`,
              waitFor,
            );
          },
        );
      },
    );
  });
});
