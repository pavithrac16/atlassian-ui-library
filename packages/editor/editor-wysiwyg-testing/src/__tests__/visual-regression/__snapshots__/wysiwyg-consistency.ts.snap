// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WYSIWYG Comparison: actions 1`] = `
Object {
  "consistency": "100%",
  "divergence": 0,
}
`;

exports[`WYSIWYG Comparison: actions inside columns 1`] = `
Object {
  "consistency": "92.05%",
  "divergence": 0.0794590856144438,
}
`;

exports[`WYSIWYG Comparison: actions inside panel 1`] = `
Object {
  "consistency": "88.85%",
  "divergence": 0.11146304675716441,
}
`;

exports[`WYSIWYG Comparison: actions inside table 1`] = `
Object {
  "consistency": "94.94%",
  "divergence": 0.05063659692064745,
}
`;

exports[`WYSIWYG Comparison: blockquote 1`] = `
Object {
  "consistency": "92.29%",
  "divergence": 0.07712418300653595,
}
`;

exports[`WYSIWYG Comparison: blockquote inside columns 1`] = `
Object {
  "consistency": "84.67%",
  "divergence": 0.1533067464273845,
}
`;

exports[`WYSIWYG Comparison: blockquote inside panel 1`] = `
Object {
  "consistency": "88.85%",
  "divergence": 0.11146304675716441,
}
`;

exports[`WYSIWYG Comparison: blockquote inside table 1`] = `
Object {
  "consistency": "79.06%",
  "divergence": 0.20940383506343713,
}
`;

exports[`WYSIWYG Comparison: bullet list 1`] = `
Object {
  "consistency": "99.98%",
  "divergence": 0.0001513840830449827,
}
`;

exports[`WYSIWYG Comparison: bullet list inside columns 1`] = `
Object {
  "consistency": "97.23%",
  "divergence": 0.027658045977011495,
}
`;

exports[`WYSIWYG Comparison: bullet list inside panel 1`] = `
Object {
  "consistency": "99.99%",
  "divergence": 0.00014450867052023122,
}
`;

exports[`WYSIWYG Comparison: bullet list inside table 1`] = `
Object {
  "consistency": "98.31%",
  "divergence": 0.016903456297773345,
}
`;

exports[`WYSIWYG Comparison: codeblock 1`] = `
Object {
  "consistency": "97.87%",
  "divergence": 0.021318527410964384,
}
`;

exports[`WYSIWYG Comparison: codeblock inside columns 1`] = `
Object {
  "consistency": "96.11%",
  "divergence": 0.038887347391786904,
}
`;

exports[`WYSIWYG Comparison: codeblock inside panel 1`] = `
Object {
  "consistency": "99.99%",
  "divergence": 0.00014450867052023122,
}
`;

exports[`WYSIWYG Comparison: codeblock inside table 1`] = `
Object {
  "consistency": "97.02%",
  "divergence": 0.02977941176470588,
}
`;

exports[`WYSIWYG Comparison: date 1`] = `
Object {
  "consistency": "98.79%",
  "divergence": 0.012096774193548387,
}
`;

exports[`WYSIWYG Comparison: date inside columns 1`] = `
Object {
  "consistency": "94.1%",
  "divergence": 0.05897435897435897,
}
`;

exports[`WYSIWYG Comparison: date inside panel 1`] = `
Object {
  "consistency": "99%",
  "divergence": 0.01001589825119237,
}
`;

exports[`WYSIWYG Comparison: date inside table 1`] = `
Object {
  "consistency": "94.9%",
  "divergence": 0.0509640522875817,
}
`;

exports[`WYSIWYG Comparison: decisions 1`] = `
Object {
  "consistency": "100%",
  "divergence": 0,
}
`;

exports[`WYSIWYG Comparison: decisions inside columns 1`] = `
Object {
  "consistency": "91.57%",
  "divergence": 0.08433677926616191,
}
`;

exports[`WYSIWYG Comparison: decisions inside panel 1`] = `
Object {
  "consistency": "99%",
  "divergence": 0.01001589825119237,
}
`;

exports[`WYSIWYG Comparison: decisions inside table 1`] = `
Object {
  "consistency": "94.77%",
  "divergence": 0.05232925384919068,
}
`;

exports[`WYSIWYG Comparison: divider 1`] = `
Object {
  "consistency": "100%",
  "divergence": 0,
}
`;

exports[`WYSIWYG Comparison: divider inside columns 1`] = `
Object {
  "consistency": "100%",
  "divergence": 0,
}
`;

exports[`WYSIWYG Comparison: divider inside panel 1`] = `
Object {
  "consistency": "99%",
  "divergence": 0.01001589825119237,
}
`;

exports[`WYSIWYG Comparison: divider inside table 1`] = `
Object {
  "consistency": "97.32%",
  "divergence": 0.0268048128342246,
}
`;

exports[`WYSIWYG Comparison: emoji 1`] = `
Object {
  "consistency": "98.68%",
  "divergence": 0.01318785578747628,
}
`;

exports[`WYSIWYG Comparison: emoji inside columns 1`] = `
Object {
  "consistency": "91.27%",
  "divergence": 0.08729260935143288,
}
`;

exports[`WYSIWYG Comparison: emoji inside panel 1`] = `
Object {
  "consistency": "98.9%",
  "divergence": 0.010969793322734499,
}
`;

exports[`WYSIWYG Comparison: emoji inside table 1`] = `
Object {
  "consistency": "93.87%",
  "divergence": 0.061274509803921566,
}
`;

exports[`WYSIWYG Comparison: heading 1`] = `
Object {
  "consistency": "98.92%",
  "divergence": 0.010814203730272597,
}
`;

exports[`WYSIWYG Comparison: heading inside columns 1`] = `
Object {
  "consistency": "91.31%",
  "divergence": 0.0868976846057572,
}
`;

exports[`WYSIWYG Comparison: heading inside panel 1`] = `
Object {
  "consistency": "99.04%",
  "divergence": 0.009599224305106659,
}
`;

exports[`WYSIWYG Comparison: heading inside table 1`] = `
Object {
  "consistency": "94.9%",
  "divergence": 0.05102441462021702,
}
`;

exports[`WYSIWYG Comparison: mention 1`] = `
Object {
  "consistency": "99.29%",
  "divergence": 0.007068311195445921,
}
`;

exports[`WYSIWYG Comparison: mention inside columns 1`] = `
Object {
  "consistency": "94.95%",
  "divergence": 0.050490196078431374,
}
`;

exports[`WYSIWYG Comparison: mention inside panel 1`] = `
Object {
  "consistency": "99.42%",
  "divergence": 0.005802861685214627,
}
`;

exports[`WYSIWYG Comparison: mention inside table 1`] = `
Object {
  "consistency": "95.33%",
  "divergence": 0.04666666666666667,
}
`;

exports[`WYSIWYG Comparison: ordered list 1`] = `
Object {
  "consistency": "99.98%",
  "divergence": 0.0001513840830449827,
}
`;

exports[`WYSIWYG Comparison: ordered list inside columns 1`] = `
Object {
  "consistency": "96.92%",
  "divergence": 0.030802062204192023,
}
`;

exports[`WYSIWYG Comparison: ordered list inside panel 1`] = `
Object {
  "consistency": "99.99%",
  "divergence": 0.00014450867052023122,
}
`;

exports[`WYSIWYG Comparison: ordered list inside table 1`] = `
Object {
  "consistency": "98.16%",
  "divergence": 0.01843635759388501,
}
`;

exports[`WYSIWYG Comparison: panel 1`] = `
Object {
  "consistency": "100%",
  "divergence": 0,
}
`;

exports[`WYSIWYG Comparison: paragraph 1`] = `
Object {
  "consistency": "92.83%",
  "divergence": 0.07167755991285403,
}
`;

exports[`WYSIWYG Comparison: paragraph inside columns 1`] = `
Object {
  "consistency": "83.65%",
  "divergence": 0.16347402597402597,
}
`;

exports[`WYSIWYG Comparison: paragraph inside panel 1`] = `
Object {
  "consistency": "93.61%",
  "divergence": 0.06387254901960784,
}
`;

exports[`WYSIWYG Comparison: paragraph inside table 1`] = `
Object {
  "consistency": "79.77%",
  "divergence": 0.20230320069204152,
}
`;

exports[`WYSIWYG Comparison: status 1`] = `
Object {
  "consistency": "99.33%",
  "divergence": 0.006688804554079697,
}
`;

exports[`WYSIWYG Comparison: status inside columns 1`] = `
Object {
  "consistency": "88.85%",
  "divergence": 0.11146304675716441,
}
`;

exports[`WYSIWYG Comparison: status inside panel 1`] = `
Object {
  "consistency": "99.49%",
  "divergence": 0.005127186009538951,
}
`;

exports[`WYSIWYG Comparison: status inside table 1`] = `
Object {
  "consistency": "94.39%",
  "divergence": 0.05607843137254902,
}
`;

exports[`WYSIWYG Comparison: table 1`] = `
Object {
  "consistency": "95.01%",
  "divergence": 0.04986787683823529,
}
`;
